### PR TITLE
Render integration test results inline.

### PR DESCRIPTION
**Pull request by michaelkaye**
Render integration test results inline (including timing information for each test), while preserving the formatting for the PR comment integration.

### CODE REVIEW

**Review by bmarty — COMMENTED** [review]
Could be nice to have a review from @ariskotsomitopoulos :)

**Comment by ariskotsomitopoulos** [comment]
@michaelkaye  Nice work, it's more clear and scalable to have the logic in a separate python file, so even if we have one large run it will still work. I can see the error `FileNotFoundError` but this is expected due to the current error/fail of the integration tests

**Review by ariskotsomitopoulos — APPROVED** [review]

**Comment by michaelkaye** [comment]
Rebased against develop so we can see the actual test run in this PR before merging.

**Comment by michaelkaye** [comment]
```
  File "./tools/ci/render_test_output.py", line 20, in <module>
    passed = int(root.attrib['passed'])
KeyError: 'passed'
```

Glad I ran this again. Now i'm confused why this is doing this.

A xml file i have locally shows:
```
<testsuite name="im.vector.app.ui.UiAllScreensSanityTest" tests="1" failures="1" errors="0" skipped="0" time="66.418" timestamp="2022-02-14T11:31:50" hostname="localhost">
```

as the line that is being parsed by the sed, and the sed definitely produces a result that has `passed=` in it, so there must be a way for the testsuite XML to contain the attribute `passed=`, but when we test in our python there isn't. WTF?

I'm going to fall back to calculating passed based on the other fields, and return that.

**Comment by michaelkaye** [comment]
> ```
>   File "./tools/ci/render_test_output.py", line 20, in <module>
>     passed = int(root.attrib['passed'])
> KeyError: 'passed'
> ```
> 
> Glad I ran this again. Now i'm confused why this is doing this.
> 
> A xml file i have locally shows:
> 
> ```
> <testsuite name="im.vector.app.ui.UiAllScreensSanityTest" tests="1" failures="1" errors="0" skipped="0" time="66.418" timestamp="2022-02-14T11:31:50" hostname="localhost">
> ```
> 
> as the line that is being parsed by the sed, and the sed definitely produces a result that has `passed=` in it, so there must be a way for the testsuite XML to contain the attribute `passed=`, but when we test in our python there isn't. WTF?
> 
> I'm going to fall back to calculating passed based on the other fields, and return that.

I've just worked out what's actually going on here, I was just not understanding how it was meant to work --

the passed= field was coming from the echo of the original script, not from the XML - so one part of me was reading it as "the thing you pass to the output, so it's passed=<body>", the other part of me was reading the end result and going "ah, that's a value called passed".

All worked out how, hopefully the next run will fix it.

**Comment by michaelkaye** [comment]
"Code Quality Checks / Kotlin Linter (pull_request) Failing after 1m — Kotlin Linter" is failing but i don't think we touched it in this change. It seems generally stable, so unsure why it's unhappy now.

https://buildkite.com/matrix-dot-org/element-android/builds/6383#2b3b62e4-a499-4cc4-867e-9e41bd29bfee also failing on buildkite for similar reason. Perhaps upstream are having issues or the package has been deleted?

**Comment by michaelkaye** [comment]
I was trying to find out where test runs had gone; but the previous version was showing "passed=<total tests of all results>" and is now showing "passed=<successful tests>" , so previously 
```
[org.matrix.android.sdk.session]
passed="25" failures="16" errors="0" skipped="2"
```
Is replaced by:
```
[org.matrix.android.sdk.session]
passed=6 failures=17 errors=0 skipped=2
```

`25 = 17 + 6 + 2` which is where the original passed=25 came from, I believe.

File: tools/ci/render_test_output.py
Comment (ouchadam) [inline tools/ci/render_test_output.py:15]: for the future, this could make a fantastic independent action

**Review by ouchadam — APPROVED** [review]
great catch with the test output counts! :100:

**Pull request merged**
